app/knights/knights.py

```diff
@@ -0,0 +1,62 @@
+from app.knights.armour import Armour
+from app.knights.weapon import Weapon
+from app.knights.potion import Potion
+
+
+class Knight:
+
+    def __init__(self, base_equip: dict) -> None:
+        self.name = base_equip.get("name")
+        self.base_power = base_equip.get("power")
+        self.base_hp = base_equip.get("hp")
+        self.protection = 0
```

Comment:
Rename this attribute to `self.base_protection` for better referring during the loop